pkg/state/degraphql_route_test.go

```diff
@@ -0,0 +1,277 @@
+					Service: &kong.Service{
+						ID: kong.String("some-service"),
+					},
+					Methods: kong.StringSlice("GET", "POST"),
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "returns an error when the degraphql route already exists",
+			degraphqlRoute: DegraphqlRoute{
+				DegraphqlRoute: kong.DegraphqlRoute{
+					ID:    kong.String("first"),
+					URI:   kong.String("/foo"),
+					Query: kong.String("query { hello }"),
+					Service: &kong.Service{
+						ID: kong.String("some-service"),
+					},
+					Methods: kong.StringSlice("GET"),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "returns an error if an id is not provided",
+			degraphqlRoute: DegraphqlRoute{
+				DegraphqlRoute: kong.DegraphqlRoute{
+					URI:   kong.String("/foo"),
+					Query: kong.String("query { hello }"),
+					Service: &kong.Service{
+						ID: kong.String("some-service"),
+					},
+					Methods: kong.StringSlice("GET"),
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "returns an error if an empty degraphql route is provided",
+			degraphqlRoute: DegraphqlRoute{
+				DegraphqlRoute: kong.DegraphqlRoute{},
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := collection.Add(tt.degraphqlRoute); (err != nil) != tt.wantErr {
+				t.Errorf("DegraphqlRoutesCollection.Add() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestDegraphqlRouteGet(t *testing.T) {
+	assert := assert.New(t)
+	collection := degraphqlRoutesCollection()
+
+	var degraphqlRoute DegraphqlRoute
+	degraphqlRoute.ID = kong.String("example")
+	degraphqlRoute.URI = kong.String("/foo")
+	degraphqlRoute.Query = kong.String("query { hello }")
+	degraphqlRoute.Service = &kong.Service{
+		ID: kong.String("some-service"),
+	}
+	degraphqlRoute.Methods = kong.StringSlice("GET")
+
+	err := collection.Add(degraphqlRoute)
+	require.Nil(t, err, "error adding degraphql route")
+
+	// Fetch the currently added entity
+	res, err := collection.Get("example")
+	require.Nil(t, err, "error getting degraphql route")
+	assert.NotNil(res)
+	assert.Equal("example", *res.ID)
+	assert.Equal("some-service", *res.Service.ID)
+
+	// Fetch non-existent entity
+	res, err = collection.Get("does-not-exist")
+	assert.NotNil(err)
+	assert.Nil(res)
+}
+
+func TestDegraphqlRouteUpdate(t *testing.T) {
+	assert := assert.New(t)
+	collection := degraphqlRoutesCollection()
+
+	var degraphqlRoute DegraphqlRoute
+	degraphqlRoute.ID = kong.String("example")
+	degraphqlRoute.URI = kong.String("/foo")
+	degraphqlRoute.Query = kong.String("query { hello }")
+	degraphqlRoute.Service = &kong.Service{
+		ID: kong.String("some-service"),
+	}
+	degraphqlRoute.Methods = kong.StringSlice("GET")
+
+	err := collection.Add(degraphqlRoute)
+	require.Nil(t, err, "error adding degraphql route")
+
+	// Fetch the currently added entity
+	res, err := collection.Get("example")
+	require.Nil(t, err, "error getting degraphql route")
+	assert.NotNil(res)
```

Comment:
This requires adding `require` instantiation above

```suggestion
	require.NotNil(res)
```